ext/ManoptJuMPExt.jl

```diff
@@ -0,0 +1,408 @@
+    # Sense of the optimization, e.g., min, max or no objective
+    sense::MOI.OptimizationSense
+    # Model used to compute gradient of the objective function with AD
+    nlp_model::MOI.Nonlinear.Model
+    # Solver parameters set with `MOI.RawOptimizerAttribute`
+    options::Dict{String,Any}
+    function Optimizer()
+        return new(
+            nothing,
+            nothing,
+            nothing,
+            Union{Nothing,Float64}[],
+            MOI.FEASIBILITY_SENSE,
+            MOI.Nonlinear.Model(),
+            Dict{String,Any}(DESCENT_STATE_TYPE => Manopt.GradientDescentState),
+        )
+    end
+end
+
+MOI.get(::Optimizer, ::MOI.SolverVersion) = "0.4.37"
+
+function MOI.is_empty(model::Optimizer)
+    # TODO replace `isnothing(model.nlp_model.objective)`
+    #      by `MOI.is_empty` once the following is fixed
+    #      https://github.com/jump-dev/MathOptInterface.jl/issues/2302
+    return isnothing(model.manifold) &&
+           isempty(model.variable_primal_start) &&
+           isnothing(model.nlp_model.objective) &&
+           model.sense == MOI.FEASIBILITY_SENSE
+end
+
+"""
+    MOI.empty!(model::ManoptJuMPExt.Optimizer)
+
+Clear all model data from `model` but keep the `options` set.
+"""
+function MOI.empty!(model::Optimizer)
+    model.manifold = nothing
+    model.problem = nothing
+    model.state = nothing
+    empty!(model.variable_primal_start)
+    model.sense = MOI.FEASIBILITY_SENSE
+    # TODO replace by `MOI.empty!` once the following is fixed
+    #      https://github.com/jump-dev/MathOptInterface.jl/issues/2302
+    model.nlp_model.objective = nothing
+    return nothing
+end
+
+"""
+    MOI.supports(::Optimizer, attr::MOI.RawOptimizerAttribute)
+
+Return a `Bool` indicating whether `attr.name` is a valid option name
+for `Manopt`.
+"""
+function MOI.supports(::Optimizer, ::MOI.RawOptimizerAttribute)
+    # FIXME Ideally, this should only return `true` if it is a valid keyword argument for
+    #       one of the `...DescentState()` constructors. Is there an easy way to check this ?
+    #       Does it depend on the different solvers ?
+    return true
+end
+
+"""
+    MOI.get(model::Optimizer, attr::MOI.RawOptimizerAttribute)
+
+Return last `value` set by `MOI.set(model, attr, value)`.
+"""
+function MOI.get(model::Optimizer, attr::MOI.RawOptimizerAttribute)
+    if !MOI.supports(model, attr)
+        throw(MOI.UnsupportedAttribute(attr))
+    end
+    return model.options[attr.name]
+end
+
+"""
+    MOI.get(model::Optimizer, attr::MOI.RawOptimizerAttribute)
+
+Set the value for the keyword argument `attr.name` to give for the constructor
+`model.options[DESCENT_STATE_TYPE]`.
+"""
+function MOI.set(model::Optimizer, attr::MOI.RawOptimizerAttribute, value)
+    if !MOI.supports(model, attr)
+        throw(MOI.UnsupportedAttribute(attr))
+    end
+    model.options[attr.name] = value
+    return nothing
+end
+
+function MOI.get(model::Optimizer, ::MOI.SolverName)
```

Comment:
This seems to be a very obvious function, but would still be nice to have a short documentation (maybe with a link to the original interface function where that is documented?)